holoviews/element/selection.py

```diff
@@ -36,6 +36,47 @@ def _get_selection_expr_for_stream_value(self, **kwargs):
     def _merge_regions(region1, region2, operation):
         return None
 
+def spatial_select_gridded(xvals, yvals, geometry):
+    rectilinear = (np.diff(xvals, axis=0) == 0).all()
+    if rectilinear:
+        from .raster import Image
+        from .path import Polygons
+        try:
+            from ..operation.datashader import rasterize
+        except ImportError:
+            raise ImportError("Lasso selection on gridded data requires "
+                              "datashader to be available.")
+        xs, ys = xvals[0], yvals[:, 0]
+        target = Image((xs, ys, np.empty(ys.shape+xs.shape)))
+        poly = Polygons([geometry])
+        mask = rasterize(poly, target=target, dynamic=False, aggregator='any')
+        return mask.dimension_values(2, flat=False)
+    else:
+        mask = spatial_select_columnar(xvals.flatten(), yvals.flatten(), geometry)
+        return mask.reshape(xvals.shape)
```

Comment:
From this implementation it looks like lasso selection on curvilinear quadmeshes is going to be slow? If so the docs should warn that. I'm curious why it can't use datashader for both rectilinear and curvilinear grids, since datashader supports both.

Reply:
We actually apply the rasterization on the lasso polygon so what I was worried about was applying the resulting mask to the quadmesh but actually there's no issue with that either.  So yes we can optimize this.

Reply:
Actually no that isn't valid, datashader can only generate a curvilinear grid from the lasso polygon which won't correctly mask the curvilinear grid.

Reply:
You mean, datashader can only generate a _rectilinear_ (and indeed equally spaced, and not just recilinear, i.e. raster) grid?  Good point; it only supports raster output grids.

Reply:
Yes, that's right fixed.